### PR TITLE
Make rename detection consistent for duplicate file ids

### PR DESCRIPTION
**Pull request by ckamm**
See #6212 and the commit message of the second commit.

### CODE REVIEW

File: src/csync/csync_private.h [hunk 1]
```diff
@@ -102,6 +102,8 @@ class ByteArrayRef
 
     friend bool operator==(const ByteArrayRef &a, const ByteArrayRef &b)
     { return a.size() == b.size() && qstrncmp(a.data(), b.data(), a.size()) == 0; }
+
+    QByteArray toByteArray() const { return QByteArray(data()); }
```
Comment (ogoffart) [inline src/csync/csync_private.h]: :-1:  You need to pass the _size as well.

You don't use this, so just remove this

File: src/csync/csync.cpp [hunk 2]
```diff
@@ -314,6 +314,9 @@ int csync_s::reinitialize() {
   local.files.clear();
   remote.files.clear();
 
+  renames.folder_renamed_from.clear();
+  renames.folder_renamed_to.clear();
```
Comment (ogoffart) [inline src/csync/csync.cpp]: :+1:

**Comment by ogoffart** [comment]
Also, can more test be added? For both commit.  Is the fact that we don't clear not causing bugs when the files are renamed back ?

**Comment by ckamm** [comment]
@ogoffart Updated and added a test for the second commit.

**Pull request merged**